controllers/addon_controller.go

```diff
@@ -158,6 +163,10 @@ func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request,
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	if addon.IsDuplicate(instance, r.versionCache) {
```

Comment:
This will cause infinite reconcile, duplicate addons should be caught (ideally) by the addon_validator which does have a check for this. 
https://github.com/keikoproj/addon-manager/blob/c3ff3e626aa81547802fb051e009da6d6aed05db/pkg/addon/addon_validator.go#L60

Reply:
This provides efficiency, reduces reconcile time - throw duplicate at earliest time. The addon_validator checks all other details though.

Reply:
This short-circuits the validator which also sets `ValidationError` state on the Addon, so this change would infinitely loop without a resolution; we need to remove this code.

Reply:
I will re-test and see if can have a better solution.

Reply:
Current issue is we always add new addon into cache w/o taking care of processAddon  errors. https://github.com/keikoproj/addon-manager/pull/96/files/e1aac607f5312ca8d2640962a82cd84b97ab67b4#diff-2df6a56a33260adb76f704ff7c6737be9f43c13fe523521ee76390ff2a9ca4b8R171-R174. 

Believing this cover some corner case.  But, if people purposely flush different addons (only name different, pkgName/Version is the same) into manager, we allow https://github.com/keikoproj/addon-manager/blob/master/pkg/addon/addon_version_cache.go#L63, which messed up the right Addon. 

This duplicateFilter at the first place try to cover this case.

Reply:
Yes, this should be the graceful solution.
c7040b3703b64edfc9e86289b906b6163387394a